tools/src/launcher/Main.hs

```diff
@@ -517,11 +532,47 @@ spawnNode (path, args, mbLogPath) = do
             logInfo "Node has started"
             return (ph, asc)
 
-runWallet :: (FilePath, [Text]) -> M ExitCode
-runWallet (path, args) = do
-    logNotice "Starting the wallet"
-    view _1 <$>
-        liftIO (readProcessWithExitCode path (map toString args) mempty)
+runWallet :: Bool -> (FilePath, [Text], Maybe FilePath) -> M ExitCode
+runWallet shouldLog (path, args, mLogPath) = do
+    customLogger LInfo NWallet "Starting the wallet"
+    phvar <- newEmptyMVar
+    liftIO $ case mLogPath of
+        Just lp -> do
+            createDirectoryIfMissing True (directory lp)
+            (_, logHandle) <- (lp,) <$> openFile lp AppendMode
+            liftIO $ IO.hSetBuffering logHandle IO.LineBuffering
+            let cr = (Process.proc (toString path) (map toString args))
+                         { Process.std_in  = Process.CreatePipe
+                         , Process.std_out = Process.UseHandle logHandle
+                         , Process.std_err = Process.UseHandle logHandle
+                         }
+            system' phvar cr mempty
+        Nothing ->
+           if shouldLog then
+               liftIO $ do
+               let cr = (Process.proc (toString path) (map toString args))
+                         { Process.std_in  = Process.CreatePipe
+                         , Process.std_out = Process.NoStream
```

Comment:
It seems you should have used `Inherit` instead of `NoStream` because it's case when wallet's logs must go into launcher's stdout/stderr

Reply:
Done